module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -185,8 +185,9 @@ namespace module::localisation {
             "Reset Robot Hypotheses",
             [this](const ResetRobotHypotheses& locReset, const Sensors& sensors) {
                 if (locReset.hypotheses.empty()) {
-                    filter.set_state(config.start_state,
-                                     std::vector<Eigen::Vector3d>(config.start_state.size(), config.start_variance));
+                    filter.set_state(
+                        config.start_state,
+                        std::vector<Eigen::Matrix3d>(config.start_state.size(), config.start_variance.asDiagonal()));
```

Comment:
what's the benefit of diagonal?

Reply:
covariance matrices are square, symmetric matrices. We generally only set the main diagonal elements because determining the off-diagonals is painful. Using `asDiagonal` takes a vector and makes a matrix that contains that vector as its main diagonal